app/utils/simplified-account/format/format-card-types.js

```diff
@@ -0,0 +1,80 @@
+const cardsThatNeedToBeEnabledOnWorldpay = ['American Express', 'Union Pay']
+
+const formatLabel = (card) => {
+  if (card.brand === 'visa' || card.brand === 'master-card') {
+    return `${card.label} ${card.type.toLowerCase()}`
+  }
+  return card.brand === 'jcb' ? card.label.toUpperCase() : card.label
+}
+
+const createCardTypeChecklistItem = (card, acceptedCards) => {
+  return {
+    value: card.id,
+    text: formatLabel(card),
+    checked: acceptedCards.filter(accepted => accepted.id === card.id).length !== 0,
+    requires3ds: card.requires3ds
+  }
+}
+
+const disableCheckboxIf3dsRequiredButNotEnabled = (cardTypeChecklistItem, accountType, accountRequires3ds) => {
+  if (cardTypeChecklistItem.requires3ds && !accountRequires3ds) {
+    return {
+      ...cardTypeChecklistItem,
+      disabled: true,
+      hint: {
+        html: accountType === 'test' ? `${cardTypeChecklistItem.text} is not available on test accounts` : `${cardTypeChecklistItem.text} cannot be used because 3D Secure is not available. Please contact support`
+      }
+    }
+  }
+  return cardTypeChecklistItem
+}
+
+const addHintForAmexAndUnionpayIfWorldpay = (cardTypeChecklistItem, paymentProvider) => {
+  if (cardsThatNeedToBeEnabledOnWorldpay.includes(cardTypeChecklistItem.text) && paymentProvider === 'worldpay') {
+    return {
+      ...cardTypeChecklistItem,
+      hint: {
+        html: 'You must have already enabled this with Worldpay'
+      }
+    }
+  }
+  return cardTypeChecklistItem
+}
+
+const formatCardTypesForAdminTemplate = (allCards, acceptedCards, account) => {
+  const debitCardChecklistItems = allCards.filter(card => card.type === 'DEBIT')
+    .map(card => createCardTypeChecklistItem(card, acceptedCards))
+    .map(cardTypeChecklistItem => disableCheckboxIf3dsRequiredButNotEnabled(cardTypeChecklistItem, account.type, account.requires3ds))
+
+  const creditCardChecklistItems = allCards.filter(card => card.type === 'CREDIT')
+    .map(card => createCardTypeChecklistItem(card, acceptedCards))
+    .map(cardTypeChecklistItem => addHintForAmexAndUnionpayIfWorldpay(cardTypeChecklistItem, account.paymentProvider))
+
+  return { debitCards: debitCardChecklistItems, creditCards: creditCardChecklistItems }
+}
+
+const formatCardTypesForNonAdminTemplate = (allCards, acceptedCards) => {
```

Comment:
the best I could come up with for refactoring this is something like this, but I'm not convinced it's an improvement so feel free to ignore it and stick with this implementation

```suggestion
const isDisabled = (card, account) => {
  return card.requires3ds && !account.requires3ds
}

const hintText = (card, account) => {
  const cardLabel = formatLabel(card)
  const disabled = isDisabled(card, account)
  if (disabled && account.type === 'test') {
    return {
      text: `${cardLabel} is not available on test accounts`
    }
  } else if (disabled) {
    return {
      text: `${cardLabel} cannot be used because 3D Secure is not available. Please contact support`
    }
  }

  if (card.type !== 'CREDIT' || account.paymentProvider !== 'worldpay') {
    return undefined
  }

  if (['American Express', 'Union Pay'].includes(cardLabel)) {
    return {
      html: 'You must have already enabled this with Worldpay'
    }
  }

  return undefined
}

const formatCard = (card, accepted, account) => {
  const disabled = isDisabled(card, account)
  return {
    value: card.id,
    text: formatLabel(card),
    checked: !disabled && accepted,
    requires3ds: card.requires3ds,
    disabled,
    hint: hintText(card, account)
  }
}

const formatCardTypesForNonAdminTemplate = (allCards, acceptedCards) => {
  const items = allCards.reduce((acc, card) => {
    if (!acc[card.type]) {
      acc[card.type] = []
    }
    const accepted = acceptedCards.filter(c => c.type === card.type && c.id === card.id).length > 0
    acc[card.type].push(formatCard(card, accepted, account))

    return acc
  }, {})

  return {
    debitCards: Object.values(items.DEBIT),
    creditCards: Object.values(items.CREDIT)
  }
}
```